networkx/algorithms/centrality/second_order.py

```diff
@@ -38,6 +38,7 @@
 __all__ = ["second_order_centrality"]
 
 
+@nx._dispatch(preserve_edge_attrs={"G": {"weight": 1}})
```

Comment:
`preserve_edge_attrs={"G": {"weight": 1}}` means that the "weight" attribute of graph G should be preserved (with default value of 1).

I think functions like this are great candidates for adding `weight="weight"` arguments rather than always assume the graph has edge attributes of `"weight"`. If `weight` argument was added, the dispatch decorator would become:
```python
@nx._dispatch(edge_attrs="weight")
```

Reply:
Find these cases on this PR with:
```bash
git grep -A 2 '_dispatch.*preserve_edge_attrs={"G":'
```
which finds:
- `second_order_centrality`
- `quotient_graph`
- `panther_similarity`
- `generate_random_paths`
- `number_of_walks`

Reply:
Raised in separate issue here: #6753